packages/expo-cli/src/commands/eas-build/utils/expoUpdates.ts

```diff
@@ -0,0 +1,370 @@
+    const result = await AndroidConfig.Updates.setUpdatesConfig(exp, androidManifestJSON, username);
+
+    await AndroidConfig.Manifest.writeAndroidManifestAsync(androidManifestPath, result);
+  }
+}
+
+async function isUpdatesConfiguredAndroidAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null
+) {
+  const buildGradlePath = getAndroidBuildGradlePath(projectDir);
+  const buildGradleContent = await getAndroidBuildGradleContentAsync(buildGradlePath);
+
+  if (!hasBuildScriptApply(buildGradleContent)) {
+    return false;
+  }
+
+  const androidManifestPath = getAndroidManifestPath(projectDir);
+  const androidManifestJSON = await AndroidConfig.Manifest.readAndroidManifestAsync(
+    androidManifestPath
+  );
+
+  if (!isMetadataSetAndroid(androidManifestJSON, exp, username)) {
+    return false;
+  }
+
+  return true;
+}
+
+function getAndroidBuildGradlePath(projectDir: string) {
+  const buildGradlePath = path.join(projectDir, 'android', 'app', 'build.gradle');
+
+  return buildGradlePath;
+}
+
+async function getAndroidBuildGradleContentAsync(buildGradlePath: string) {
+  if (!(await fs.pathExists(buildGradlePath))) {
+    throw new Error(`Couldn't find gradle build script at ${buildGradlePath}`);
+  }
+
+  const buildGradleContent = await fs.readFile(buildGradlePath, 'utf-8');
+
+  return buildGradleContent;
+}
+
+function hasBuildScriptApply(buildGradleContent: string): boolean {
+  return (
+    buildGradleContent
+      .split('\n')
+      // Check for both single and double quotes
+      .some(line => line === androidBuildScript || line === androidBuildScript.replace(/"/g, "'"))
+  );
+}
+
+function getAndroidManifestPath(projectDir: string) {
+  const manifestPath = path.join(
+    projectDir,
+    'android',
+    'app',
+    'src',
+    'main',
+    'AndroidManifest.xml'
+  );
+
+  return manifestPath;
+}
+
+function isMetadataSetAndroid(
+  androidManifestJSON: AndroidConfig.Manifest.Document,
+  exp: ExpoConfig,
+  username: string | null
+): boolean {
+  const currentSdkVersion = AndroidConfig.Updates.getSDKVersion(exp);
+  const currentRuntimeVersion = AndroidConfig.Updates.getRuntimeVersion(exp);
+  const currentUpdateUrl = AndroidConfig.Updates.getUpdateUrl(exp, username);
+
+  const setRuntimeVersion = getAndroidMetadataValue(
+    androidManifestJSON,
+    AndroidConfig.Updates.Config.RUNTIME_VERSION
+  );
+
+  const setSdkVersion = getAndroidMetadataValue(
+    androidManifestJSON,
+    AndroidConfig.Updates.Config.SDK_VERSION
+  );
+
+  const setUpdateUrl = getAndroidMetadataValue(
+    androidManifestJSON,
+    AndroidConfig.Updates.Config.UPDATE_URL
+  );
+
+  return Boolean(
+    (currentRuntimeVersion
+      ? setRuntimeVersion === currentRuntimeVersion
+      : currentSdkVersion && setSdkVersion === currentSdkVersion) &&
+      currentUpdateUrl &&
+      setUpdateUrl === currentUpdateUrl
+  );
+}
+
+function getAndroidMetadataValue(
```

Comment:
Maybe move this to `config/src/android/Manifest` next to the remove command you made